pkg/kudoctl/cmd/bundle.go

```diff
@@ -0,0 +1,196 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kudobuilder/kudo/pkg/bundle"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/yaml"
+)
+
+const apiVersion = "kudo.k8s.io/v1alpha1"
+
+type bundleCmd struct {
+	out        io.Writer
+	bundlePath string
+}
+
+// NewBundleCmd creates a bundle command for the CLI
+func newBundleCmd(out io.Writer) *cobra.Command {
+	bundleCmd := &cobra.Command{
+		Use:     "bundle <path>",
+		Short:   "-> Bundle a package from the Beta Framework format into FrameworkVersion (experimental)",
+		Long:    `Bundle a package from the Beta Framework format into FrameworkVersion (experimental)`,
+		Example: "kubectl kudo bundle . | kubectl apply -f -",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			bundle := &bundleCmd{
+				out: out,
+			}
+
+			if len(args) != 1 {
+				return errors.Errorf("Bundle requires 1 path argument, got %d", len(args))
+			}
+
+			bp, err := resolveBundlePath(args[0])
+			if err != nil {
+				return err
+			}
+			bundle.bundlePath = bp
+
+			return bundle.run()
+		},
+		SilenceUsage: true,
+	}
+
+	const usageFmt = "Usage:\n  %s\n\nFlags:\n%s"
+	bundleCmd.SetUsageFunc(func(cmd *cobra.Command) error {
+		fmt.Fprintf(bundleCmd.OutOrStderr(), usageFmt, bundleCmd.UseLine(), bundleCmd.Flags().FlagUsages())
+		return nil
+	})
+	return bundleCmd
+}
+
+func (b *bundleCmd) run() error {
+	frameworkPath := path.Join(b.bundlePath, "framework.yaml")
```

Comment:
would we not also need to do this for `framework.yaml` but also `params.yaml` and `common/common.yaml` ?

Reply:
I think `common/*.yaml` are all templated objects